pkg/engine/executor/vector_aggregate.go

```diff
@@ -119,7 +133,7 @@ func (v *vectorAggregationPipeline) read(ctx context.Context) (arrow.Record, err
 					labelValues[col] = arr.Value(row)
 				}
 
-				v.aggregator.Add(tsCol.Value(row).ToTime(arrow.Nanosecond), valueArr.Value(row), labelValues)
+				v.aggregator.Add(tsCol.Value(row).ToTime(arrow.Nanosecond), float64(valueArr.Value(row)), labelValues)
```

Comment:
the following line should change to `*array.Float64` as the generated value is a float now. 

https://github.com/grafana/loki/blob/c022ebe642c51ddbf97bc8b9579bd138c0ebdbef/pkg/engine/executor/vector_aggregate.go#L112

Reply:
A similar change should be done in compat.go
https://github.com/grafana/loki/blob/c022ebe642c51ddbf97bc8b9579bd138c0ebdbef/pkg/engine/compat.go#L307